### PR TITLE
Add the FWFormViewController that will compose the Form

### LINKED ISSUE #4: Main view controller to host UI Components
**What is an UI Component?** A UI Component is a representation of one of many types of inputs available in a form. It is composed of a UIViewController and a ViewModel, which communicates with its UIViewController through the Protocol-delegate pattern.
**A UI Component should only be instantiated as part of a root UIViewController through UIViewController containment.**

**What is a root UIViewController?** A root UIViewController, named in code as RootViewController, is a UIViewController responsible for being the container of all the UI Components of a given form. This object is the one responsible for the instantiation of each UI Component and for receiving and sending data to the "outside world".

This first implementation will be simple and will require refactoring, likely sooner than we think. However, establishing this primary contact with other parts of the framework will give us a better grasp of what the architecture of our codebase should be.
Ideally, each UIViewController should have a ViewModel, but for this first implementation an empty class will be enough.

### PR DESCRIPTION
**Pull request by VicFalcetta**
## Motivation
Closes #4 

## Modifications
Created three files: FWFormViewController, FWFormViewControllerCell and FormSpec. 
- FWFormViewController: It will be the VC that will hold all of the components for the form as cells of a CollectionView.
- FWFormViewControllerCell: It is the cell for the CollectionView in the main VC
- FormSpec: An enum that holds all design variables that will be used in making the form.

## Result
Now the framework has its ViewController that will compose its form.

##  Checklist
 **This PR is**
* [x] In accordance with our coding principles.
* [x] Implementing tests wherever needed and possible.
* [x] Free of commented code.
* [x] Documented.
* [x] Closing an Issue or related to an Issue
* [x] Approved in the CI

### CODE REVIEW

File: Sources/Formworks/FWForm/FWFormViewController.swift [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    //- MARK: Life cycle
+    public override func loadView() {
+        super.loadView()
+        view.backgroundColor = .fwFormBackground
+    }
+    
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        setUpCollectionViewConstraints()
+    }
```
Comment (Galdineris) [inline Sources/Formworks/FWForm/FWFormViewController.swift:42]: @VicFalcetta why did you split the controller configurations between loadView and viewDidLoad?

File: Sources/Formworks/FWForm/FWFormViewController.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+}
+//- MARK: UICollectionViewDelegate
+extension FWFormViewController: UICollectionViewDelegate {
+    
+}
+//- MARL: UICollectionViewDataSource
+extension FWFormViewController: UICollectionViewDataSource {
+    
+    public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return 30
```
Comment (Galdineris) [inline Sources/Formworks/FWForm/FWFormViewController.swift:85]: I understand this is for development purposes but don't forget to raise an issue about cell configuration later.

Comment (VicFalcetta) [reply Sources/Formworks/FWForm/FWFormViewController.swift:85]: Thanks for reminding this. I will create an issue stating that the cells need to be properly configurated.

**Review by csfar — APPROVED** [review]

File: Sources/Formworks/FWForm/FWFormViewController.swift (hunk 1 above)
Comment (VicFalcetta) [reply Sources/Formworks/FWForm/FWFormViewController.swift:42]: In this case is just to separate what is needed to set up the View Controller (such as background color) and what is needed to be set up for its subviews (in this case the collection view).

Comment (Galdineris) [reply Sources/Formworks/FWForm/FWFormViewController.swift:42]: Alright. I get it now.
Thanks for the clarification!

**Review by Galdineris — APPROVED** [review]
All good, very well done.

**Pull request merged**